crossbar/webservice/static.py

```diff
@@ -131,7 +131,7 @@ def create(transport, path, config):
 
             raise ApplicationError("crossbar.error.invalid_configuration", "missing web spec")
 
-        static_dir = static_dir.encode('ascii', 'ignore')  # http://stackoverflow.com/a/20433918/884770
+        static_dir_enc = static_dir.encode('ascii', 'ignore')  # http://stackoverflow.com/a/20433918/884770
```

Comment:
why do we need to change the variable name?

Reply:
needs to be "static_dir"

Reply:
We use the existing `static_dir` variable in the newly added code below, It was being encoded and overwritten previously. As `os.path.join()` expects both paths to be string.

The alternate could be to just avoid the renaming and call `static_dir.decode('ascii')` where needed. Your call!